added cleanup command

professor91 opened the pull request.

professor91 committed:
added cleanup command

  - admin command to reset an inactive team
    - remove all users and team lead
    - delete old and create new team text channel
    - updates db with new team text channel id

rollie42 commented on server/src/testing/testingRoutes.ts:
```diff
@@ -128,5 +128,59 @@ export default function testingHandler(server: FastifyInstance, options, done) {
     console.log(res)
     reply.code(200).send(res)
   })
+  server.get('/genData', async (req, reply) => {
```
Can we guard this so it's only accessible in non-dev env?

rollie42 replied:
Actually, we should block this entire handler in prod - so changing it in server.ts line 65 to only register in dev environment

rollie42 commented on server/src/testing/testingRoutes.ts:
```diff
@@ -128,5 +128,59 @@ export default function testingHandler(server: FastifyInstance, options, done) {
+    }
+  })
+  server.get('/testcleanup', async (req, reply) => {
+    const targetTeam = await firebaseApi.getTeam('test_role')
```
This doesn't clean up the 2nd team

professor91 replied:
didn't get it...the script is supposed to mimic the cleanupcommand which basically takes a team and resets it, it does not handle multiple teams input

rollie42 replied:
Ah I see what this is intending - okay to leave as is

rollie42 commented on server/src/hwApi.ts:
```diff
@@ -321,4 +321,38 @@ removed from your team and need to be re-added if you wish to participate in fut
+      await this.removeUserFromTeam(user)
+
+      if (user.teamLead) {
+        await this.setTeamLead(user, false)
```
removeUserFromTeam should be changed to do this

rollie42 commented on server/src/admin/commands/admin-commands/CleanUpTeam.ts:
```diff
@@ -0,0 +1,29 @@
+import { RegisteredUserAppCommand } from "../DiscordAppCommand.js";
```
Every command filename is "DoSomethingCommand.ts" - should rename this file appropriately

rollie42 commented on server/src/hwApi.ts:
```diff
@@ -321,4 +321,38 @@ removed from your team and need to be re-added if you wish to participate in fut
       message: 'Github id updated',
     }
   }
+  async cleanupTeam(teamRole: string) {
```
If you look  at the other functions in this API, they are all like:

async addUserToTeam(user: UserT, team: TeamT, silent = false) {
async removeUserFromTeam(user: UserT) {

etc. We don't use identifiers here, so instead have it accept `(team: TeamT)`

rollie42 commented on server/src/discord/discordApi.ts:
```diff
@@ -189,6 +189,17 @@ ${await teamsSection()}${await usersSection()}
     }
   }
+  
+  async resetTeamDefaultChannel(targetTeam: TeamT) {
```
Similar to how hwApi works off the types defined in the app, *this* file only works with discord "types".  Imagine if this were shared with other users as an API to access discord APIs. If you have a tic-tac-toe game, there's no concept of a 'TeamT'. Also this API shouldn't be calling things inside the firebaseAPI (I did break this rule for the LFG/LFM stuff, which perhaps should be changed as well, but it's fine to leave that part alone for your change)

I think if this file is exposing discord functionality, it's fine to have a 'recreateChannel' function that gets the category, etc, and then just returns the new channel ID, which is saved at the call site for this function, presumably in hwApi.

professor91 replied:
agreed I will update the params

about second point u mentioned `file`...iiuc `recreateChannel` should be a static method of discordApi, right?

rollie42 replied:
There aren't any static methods, it should just be a method in discordApi like any other, that takes in a channel id, and returns the newly created channel id

professor91 committed:
implemented suggestions by rollie

  - register testing endpoint only in dev
  - update removeUserFromTeam to handle teamLead status of user
  - added new function recreateChannel
